File: day_16/task_1.rb
rules = {}
tickets = []

mode = :rules

File.readlines("input.txt").each do |line|
  next mode = :ignore if line == "\n"
  next mode = :tickets if line == "nearby tickets:\n"

  if mode == :rules
    rule, ranges = line.split(": ")
    rules[rule] = ranges.split(" or ").map { |e| Range.new(*e.split("-").map(&:to_i)) }
  elsif mode == :tickets
    tickets << line.split(",").map(&:to_i)
  end
end

invalid_values = tickets.inject(0) do |sum, ticket|
  ticket.each do |value|
    next if rules.values.any? { |ranges| ranges.any? { |range| range.include?(value) } }

    sum += value
  end

  sum
end

p invalid_values

File: 2021/01/task_1.py
file = open("input.txt", "r")

previous_depth = None
increased_depth_counter = 0

for line in file.readlines():
  depth = int(line.strip("\n"))
  
  if previous_depth != None and depth > previous_depth:
    increased_depth_counter += 1 
    
  previous_depth = depth
  
print(increased_depth_counter)
